libs/webb-ui-components/src/components/BridgeInputs/AdjustAmount.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  AddCircleFillIcon,
+  AddCircleLineIcon,
+  IndeterminateCircleFillIcon,
+  IndeterminateCircleLineIcon,
+} from '@webb-tools/icons';
+import { forwardRef, useCallback, useEffect, useMemo, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { Input } from '../Input';
+import { AdjustAmountProps } from './types';
+
+/**
+ * The `AdjustAmount` component
+ *
+ * Props:
+ *
+ * - `id`: The `id` prop for label and input (defaults to "adjust-amount")
+ * - `value`: The value prop
+ * - `onChange`: The callback function to control the component
+ * - `min`: The minimum value
+ * - `max`: The maximum value
+ * - `step`: The step value (defaults to 0.5)
+ *
+ * @example
+ *
+ * ```jsx
+ * <AdjustAmount value={value} onChange={(nextVal) => setValue(nextVal)} />
+ * ```
+ */
+export const AdjustAmount = forwardRef<HTMLDivElement, AdjustAmountProps>(
+  (
+    {
+      className,
+      iconClassName,
+      id = 'adjust-amount',
+      isDisabled,
+      max,
+      min,
+      onChange,
+      step = 0.5,
+      value: valueProp = 0,
+      ...props
+    },
+    ref
+  ) => {
+    const [value, setValue] = useState(valueProp);
+
+    // Subscribes to the `value` prop
+    useEffect(() => {
+      setValue(valueProp);
+    }, [valueProp]);
+
+    const sharedIconBtnClsx = useMemo(
+      () =>
+        twMerge(
+          'group',
+          'fill-mono-160 hover:fill-mono-200 disabled:fill-mono-100',
+          'dark:fill-mono-40 dark:disabled:fill-mono-120',
+          iconClassName
+        ),
+      [iconClassName]
+    );
+
+    const handleMinusClick = useCallback(() => {
+      if (isDisabled) return;
+
+      const nextVal = value - step;
+
+      if (!min) {
+        setValue(nextVal);
+        onChange?.(nextVal);
+        return;
+      }
+
+      if (nextVal < min) {
+        return;
+      }
+
+      setValue(nextVal);
+      onChange?.(nextVal);
```

Comment:
I think this can be simplified into this:

```
if (typeof min === "number" && nextVal < min) {
  return;
}

setValue(nextVal);
onChange?.(nextVal);
```